spec/rspec/support/method_signature_verifier_spec.rb

```diff
@@ -366,6 +366,105 @@ def arity_kw(x, y:1, z:2); end
           end
         end
 
+        if RubyFeatures.kw_args_supported?
+          describe 'a method with optional argument and keyword arguments' do
+            eval <<-RUBY
+              def arity_kw(x, y = {}, z:2); end
+            RUBY
+
+            let(:test_method) { method(:arity_kw) }
+
+            it 'does not require any of the arguments' do
+              expect(valid?(nil)).to eq(true)
+              expect(valid?(nil, nil)).to eq(true)
+            end
+
+            it 'does not allow an invalid keyword arguments' do
+              expect(valid?(nil, nil, :a => 1)).to eq(false)
+              expect(valid?(nil, :a => 1)).to eq(false)
+            end
+
+            it 'treats symbols as keyword arguments and the rest as optional argument' do
+              expect(valid?(nil, 'a' => 1)).to eq(true)
+              expect(valid?(nil, 'a' => 1, :z => 3)).to eq(true)
+              expect(valid?(nil, 'a' => 1, :b => 3)).to eq(false)
+              expect(valid?(nil, 'a' => 1, :b => 2, :z => 3)).to eq(false)
+            end
+
+            it 'mentions the invalid keyword args in the error', :pending => RSpec::Support::Ruby.jruby? && !RSpec::Support::Ruby.jruby_9000? do
+              expect(error_for(nil, nil, :a => 0)).to \
+                eq("Invalid keyword arguments provided: a")
+              expect(error_for(nil, :a => 0)).to \
+                eq("Invalid keyword arguments provided: a")
+              expect(error_for(nil, 'a' => 0, :b => 0)).to \
+                eq("Invalid keyword arguments provided: b")
```

Comment:
Stylistically we prefer having these all on one line unless it hits the limits, then we'd divy it up like:
```
expect(
  ...
).to ...
```